### PR TITLE
Feature/h3ronpy

### PR DESCRIPTION
**Pull request by zacharyDez**
### What I Changed
1. **Refactored `generate_h3_geometries` function**:
    - Added support for handling H3 hexagon IDs in `uint64` format and converting them to geometries (either polygons or points).
    - Fixed the logic to correctly handle H3 ID to geometry conversion, using `cells_to_wkb_polygons` for polygons and `cells_to_wkb_points` for centroids.
    - Improved error handling by raising proper exceptions if invalid geometry types or pyarrow Scalar types are encountered.

2. **Adjusted `aggregate` and `summaries` methods in `StatsTable`**:
    - Integrated `cells_to_string` to convert H3 IDs before database operations.
    - Added a `ValueError` check if the query returns no rows (to catch invalid fields).
    - Refined SQL query generation to ensure it properly interacts with H3 IDs stored as hexagons.

3. **Unit Tests**:
    - Updated the test cases in `test_h3_utils.py` and `test_module.py` to reflect the changes in how geometries are generated.
    - Added assertions for error handling (e.g., invalid field names in `aggregate` should now raise a `ValueError`).
    - Updated tests to include both polygon and point geometry handling.

4. **Error Fixes**:
    - Resolved the `TypeError: Expected pyarrow Buffer, but got LargeBinaryScalar` by ensuring proper conversion from pyarrow Scalar types.
    - Fixed SQL query results processing to handle and print row outputs when appropriate.

---

### How to Test It
1. **Unit Tests**:
    - Run `pytest` on the following test files:
        - `tests/test_module.py`
        - `tests/test_h3_utils.py`
        - `tests/test_api.py`
    - These tests will cover:
        - H3 ID generation
        - Geometry conversion for both polygons and points
        - Database interactions via the `StatsTable` methods
        - Error handling for invalid field names and geometries

2. **Notebooks**:
    - Test the updated geometry generation functionality by running the relevant notebooks, such as `notebooks/user-docs/space2stats_api_bug.ipynb`, which shows how point geometries aren't being generated properly.

---

### Other Notes
- `notebooks/user-docs/space2stats_api_bug.ipynb` shows an example where point geometries are not being generated correctly. Further investigation is needed into the issue with point geometries and H3 ID conversion.

### CODE REVIEW

**Comment by zacharyDez** [comment]
@kylebarron; some weird behavior with points I might need help with:
image

File: space2stats_api/src/space2stats/h3_utils.py [hunk 1]
```diff
@@ -1,145 +1,79 @@
...
+    for geom in wkb_geometries:
+        if isinstance(geom, pa.LargeBinaryScalar):
+            # Convert the LargeBinaryScalar to bytes
+            shapely_geom = wkb.loads(geom.as_py())
+            # Convert Shapely geometry to GeoJSON-like dict
+            geojson_geometries.append(mapping(shapely_geom))
+        else:
+            raise TypeError(
+                f"Expected pa.LargeBinaryScalar, but got {type(geom).__name__}"
+            )
```
Comment (kylebarron) [inline space2stats_api/src/space2stats/h3_utils.py:77]: This loop will be really slow. You can instead use `shapely.from_wkb` right on the pyarrow array of WKB geometries.

```
shapely.to_geojson(shapely.from_wkb(wkb_geometries))
```

File: space2stats_api/src/pyproject.toml [hunk 2]
```diff
@@ -21,7 +21,8 @@ starlette-cramjam = ">=0.3,<0.4"
 mangum = "*"
 asgi-s3-response-middleware = "^0.0.4"
 boto3 = "^1.35.11"
-numpy = "^2.1.1"
+numpy = "^1.24.0"
```
Comment (kylebarron) [inline space2stats_api/src/pyproject.toml:24]: Hmm we should get h3ronpy updated to work with numpy v2. That depends on this PR https://github.com/PyO3/rust-numpy/pull/442

**Comment by zacharyDez** [comment]
While waiting for this to be resolved: https://github.com/nmandery/h3ronpy/issues/58, I opted to use the `cells_to_coordinates(h3_ids_uint64)` function.

File: space2stats_api/src/pyproject.toml (hunk 2 above)
Comment (zacharyDez) [reply space2stats_api/src/pyproject.toml:24]: While waiting for this to be resolved: https://github.com/nmandery/h3ronpy/issues/58, I opted to use the cells_to_coordinates(h3_ids_uint64) function.

File: space2stats_api/src/space2stats/h3_utils.py (hunk 1 above)
Comment (zacharyDez) [reply space2stats_api/src/space2stats/h3_utils.py]: ah, good catch. I'll have to update the API structure and there's a little mismatch while still use the coordinate approach. Might be worth tackling in a separate PR? Performance has improved tremedously already.

Comment (zacharyDez) [reply space2stats_api/src/space2stats/h3_utils.py]: just implemented.

**Comment by Gabe-Levin** [comment]
@zacharyDez I ran the test files specified above and they are all passing. However, I couldn't find this note book: notebooks/user-docs/space2stats_api_bug.ipynb

Has it been removed on purpose, is it still relevant?

**Comment by zacharyDez** [comment]
Yes! It's been fixed in the new release of h3ronpy, so I removed the notebook.

**Pull request merged**